api/src/test/java/ca/bc/gov/educ/studentdatacollection/api/controller/v1/CodeTableControllerTest.java

```diff
@@ -0,0 +1,158 @@
+package ca.bc.gov.educ.studentdatacollection.api.controller.v1;
+
+import ca.bc.gov.educ.studentdatacollection.api.BaseStudentDataCollectionAPITest;
+import ca.bc.gov.educ.studentdatacollection.api.StudentDataCollectionApiApplication;
+import ca.bc.gov.educ.studentdatacollection.api.constants.v1.URL;
+import ca.bc.gov.educ.studentdatacollection.api.model.v1.*;
+import ca.bc.gov.educ.studentdatacollection.api.repository.v1.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.time.LocalDateTime;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.oidcLogin;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Slf4j
+class CodeTableControllerTest extends BaseStudentDataCollectionAPITest {
+
+  @Autowired
+  private MockMvc mockMvc;
+  @Autowired
+  EnrolledProgramCodeRepository enrolledProgramCodeRepository;
+  @Autowired
+  CareerProgramCodeRepository careerProgramCodeRepository;
+  @Autowired
+  HomeLanguageSpokenCodeRepository homeLanguageSpokenCodeRepository;
+  @Autowired
+  BandCodeRepository bandCodeRepository;
+  @Autowired
+  FundingCodeRepository fundingCodeRepository;
+  @Autowired
+  EnrolledGradeCodeRepository enrolledGradeCodeRepository;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.openMocks(this);
+  }
+
+  protected final static ObjectMapper objectMapper = JsonMapper.builder().addModule(new JavaTimeModule()).build();
+
+  @BeforeEach
+  public void before() {
+    enrolledProgramCodeRepository.save(this.createEnrolledProgramCodeData());
+    careerProgramCodeRepository.save(this.createCareerProgramCodeData());
+    homeLanguageSpokenCodeRepository.save(this.homeLanguageSpokenCodeData());
+    bandCodeRepository.save(this.bandCodeData());
+    fundingCodeRepository.save(this.fundingCodeData());
+    enrolledGradeCodeRepository.save(this.enrolledGradeCodeData());
+  }
+  @AfterEach
+  public void after() {
+    enrolledProgramCodeRepository.deleteAll();
+    careerProgramCodeRepository.deleteAll();
```

Comment:
I'm going to try to convince you to put these cleanup methods into the cleanDB rest utils. 

https://github.com/bcgov/EDUC-STUDENT-DATA-COLLECTION-API/blob/master/api/src/test/java/ca/bc/gov/educ/studentdatacollection/api/support/StudentDataCollectionTestUtils.java

^^^ This runs after each test. 

Benefits will be less code duplication and guesswork to clear out our database.

Reply:
Sure, I'll update that in the next PR. I have to move entity creation methods to the base class as well